Marlin/MarlinSerial.cpp

```diff
@@ -325,144 +325,72 @@ MarlinSerial customizedSerial;
 
     enum e_parser_state {
       state_RESET,
+      state_N,
       state_M,
       state_M1,
       state_M10,
       state_M11,
-      state_M2,
-      state_M3,
       state_M4,
       state_M41,
       state_IGNORE // to '\n'
     };
 
     static e_parser_state state = state_RESET;
 
+    if (c == '\n') state = state_IGNORE;
+
     switch (state) {
       case state_RESET:
         switch (c) {
-          case 'M':
-            state = state_M;
-            break;
-          case ';':
-            state = state_IGNORE;
-            break;
-          default: state = state_RESET;
+          case ' ': break;
+          case 'N': state = state_N;      break;
+          case 'M': state = state_M;      break;
+          default: state = state_IGNORE;
         }
-      break;
+        break;
+
+      case state_N:
+        if (!NUMERIC_SIGNED(c)) state = state_RESET;
+        break;
```

Comment:
The main thing I want to ensure is that we are at **the start of the command** and not anyplace else. If (for some reason) we get a command like "G12 M108" we want to ignore the "M108" in this case. So in `state_RESET` (at the beginning of a new line) if we encounter "N" that's okay. We can skip past it and then go back to `state_RESET`. If there's a space just skip. If we see "M", then we go to `state_M`. With the added screening for "N", every other character can jump to `state_IGNORE` and stay there for the rest of the line.